packages/effector/index.d.ts

```diff
@@ -83,6 +83,7 @@ export type CompositeName = {
 
 export interface Event<Payload> extends Unit<Payload> {
   (payload: Payload): Payload
+  (this: Payload extends void ? void : `Error: Expected 1 argument, but got 0`): void
```

Comment:
What is this?

Reply:
"this" is a type of context, it is void by default
<img width="730" alt="CleanShot 2020-10-09 at 08 15 57@2x" src="https://user-images.githubusercontent.com/29141708/95545866-ba34bf80-0a07-11eb-90ca-a0ab481841b1.png">

Reply:
What is \`Error: Expected 1 argument, but got 0\`?

Reply:
It's just a text to show in the error message

Reply:
According to tests, it will be shown only if Event expects an argument, but no parameters given.